src/main/java/core/basesyntax/service/ShopServiceImpl.java

```diff
@@ -0,0 +1,33 @@
+package core.basesyntax.service;
+
+import core.basesyntax.operation.OperationHandler;
+import core.basesyntax.strategy.OperationStrategy;
+import core.basesyntax.storage.Storage;
+import core.basesyntax.model.FruitTransaction;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class ShopServiceImpl implements ShopService {
+    private final OperationStrategy operationStrategy;
+    private final Map<String, Integer> storage = Storage.getFruitStorage();
+
+    public ShopServiceImpl(OperationStrategy operationStrategy) {
+        this.operationStrategy = operationStrategy;
+    }
+
+    @Override
+    public void process(List<FruitTransaction> transactions) {
+        for (FruitTransaction transaction : transactions) {
+
+            OperationHandler handler = operationStrategy.getHandler(transaction.getOperation());
+
+            handler.apply(storage, transaction);
+        }
+    }
+
+    @Override
+    public Map<String, Integer> getStorage() {
```

Comment:
The method `getStorage()` should not be a part of the `ShopService` interface as per SOLID principles. It violates the single responsibility principle because the service should only be responsible for processing transactions, not providing direct access to the storage.